qase-javascript-commons/src/reporters/testops-reporter.ts

```diff
@@ -165,61 +177,97 @@ export class TestOpsReporter extends AbstractReporter {
   /**
    * @returns {Promise<void>}
    */
-  public async publish(): Promise<void> {
-    let runId: number;
+  public async startTestRun(): Promise<void> {
+    await this.checkOrCreateTestRun();
+  }
 
-    if (this.run.id !== undefined) {
-      await this.checkRun(this.run.id);
+  /**
+   * @param {TestResultType} result
+   * @returns {Promise<void>}
+   */
+  public override async addTestResult(result: TestResultType): Promise<void> {
+    await super.addTestResult(result);
 
-      runId = this.run.id;
-    } else {
-      const { result } = await this.createRun(
-        this.run.title,
-        this.run.description,
-        this.environment,
-      );
+    if (!this.isTestRunReady) {
+      return;
+    }
 
-      if (!result?.id) {
-        throw new Error('Cannot create run.');
-      }
+    const countOfResults = this.chunk + this.firstIndex;
 
-      runId = result.id;
+    if (this.results.length >= countOfResults) {
+      await this.publishResults(this.results.slice(this.firstIndex, countOfResults));
+      this.firstIndex = countOfResults;
     }
+  }
 
-    if (!this.results.length) {
-      this.log(
-        'No test cases were matched. Ensure that your tests are declared correctly.',
-      );
+  /**
+   * @returns {Promise<void>}
+   */
+  private async checkOrCreateTestRun(): Promise<void> {
+    if (this.run.id !== undefined) {
+      await this.checkRun(this.run.id);
 
+      this.isTestRunReady = true;
```

Comment:
What happens if we get a wrong run id?

Reply:
Oh, I see, `checkRun` throws an exception.

Reply:
yep